runtime error on bad user input

### Describe the bug

`$ gh version 0.12.0-49-gf30bc5b (2020-09-09)`

I have been using the CLI for some time and its really great :). I was looking to contribute to some open issues and stumbled upon [#1110](https://github.com/cli/cli/issues/1110). In order to reproduce the error given in the issue you have to run the following:
`$ gh -R cli/cli issue view 1110`. This leads to runtime error and an ugly message is displayed (see logs below). 

The error is quite a corner case and requires the user to give very bad input. 
### Steps to reproduce the behaviour

1. Type this:
```
$ gh -R cli/cli issue view 1110
or
$ gh -w view repo cli/cli
```


### Expected vs actual behavior

I would like a nice error message to be displayed. 

From debugging the code the error seems to be rooted here, where cmd is set to nil and the following if statement fails to catch the error.

https://github.com/cli/cli/blob/33e11f02e46339a4cab0d85da0f368bd8e8df480/cmd/gh/main.go#L53-L54

Then we try to dereference a null pointer later on:

https://github.com/cli/cli/blob/33e11f02e46339a4cab0d85da0f368bd8e8df480/cmd/gh/main.go#L95-L97

The value of `err` returned from `rootCmd.Traverse(expandedArgs)`, in the second sample case is:
`"unknown shorthand flag: 'w' in -w"`.

I propose we display something like this:
```
$ gh -w view repo cli/cli
$ failed to parse the command:  unknown shorthand flag: 'w' in -w
```

I think this only requires a small fix and I'm going to create a pull request with my proposed changes after posting this. 

### Logs

```
cli/cli $ gh -R cli/cli issue view 1110
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x110 pc=0x1734456]

goroutine 1 [running]:
github.com/spf13/cobra.(*Command).Runnable(...)
	github.com/spf13/cobra@v1.0.0/command.go:1313
github.com/cli/cli/pkg/cmdutil.IsAuthCheckEnabled(0x0, 0xc0002be430)
	github.com/cli/cli/pkg/cmdutil/auth_check.go:36 +0x26
main.main()
	github.com/cli/cli/cmd/gh/main.go:95 +0x63a
```

 ```
cli/cli $ gh -w view repo cli/cli
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x110 pc=0x174da86]

goroutine 1 [running]:
github.com/spf13/cobra.(*Command).Runnable(...)
	github.com/spf13/cobra@v1.0.0/command.go:1313
github.com/cli/cli/pkg/cmdutil.IsAuthCheckEnabled(0x0, 0x17)
	github.com/cli/cli/pkg/cmdutil/auth_check.go:36 +0x26
main.main()
	github.com/cli/cli/cmd/gh/main.go:97 +0xe36
 ```

--- a/cmd/gh/main.go
+++ b/cmd/gh/main.go
@@ -51,11 +51,7 @@ func main() {
 	}
 
 	cmd, _, err := rootCmd.Traverse(expandedArgs)
-	if err != nil {
-		fmt.Fprintf(stderr, "failed to parse the command:  %s\n", err)
-		os.Exit(2)
-	}
-	{
+	if err != nil || cmd == rootCmd {
 		originalArgs := expandedArgs
 		isShell := false
 
@@ -94,6 +90,11 @@ func main() {
 
 			os.Exit(0)
 		}
+	}
+
+	if cmd == nil {
+		fmt.Fprintf(stderr, "failed to parse command, %s\n", err)
+		os.Exit(2)
 	}
 
 	authCheckEnabled := os.Getenv("GITHUB_TOKEN") == "" &&

--- a/cmd/gh/main.go
+++ b/cmd/gh/main.go
@@ -51,7 +51,11 @@ func main() {
 	}
 
 	cmd, _, err := rootCmd.Traverse(expandedArgs)
-	if err != nil || cmd == rootCmd {
+	if err != nil {
+		fmt.Fprintf(stderr, "failed to parse the command:  %s\n", err)
+		os.Exit(2)
+	}
+	{
 		originalArgs := expandedArgs
 		isShell := false
 

--- a/cmd/gh/main.go
+++ b/cmd/gh/main.go
@@ -92,14 +92,9 @@ func main() {
 		}
 	}
 
-	if cmd == nil {
-		fmt.Fprintf(stderr, "failed to parse command, %s\n", err)
-		os.Exit(2)
-	}
-
 	authCheckEnabled := os.Getenv("GITHUB_TOKEN") == "" &&
 		os.Getenv("GITHUB_ENTERPRISE_TOKEN") == "" &&
-		cmdutil.IsAuthCheckEnabled(cmd)
+		cmd != nil && cmdutil.IsAuthCheckEnabled(cmd)
 	if authCheckEnabled {
 		cfg, err := cmdFactory.Config()
 		if err != nil {

